File: yangxj96-serve/yangxj96-serve-auth/src/main/java/com/yangxj96/saas/server/auth/service/AuthService.java
package com.yangxj96.saas.server.auth.service;

import com.yangxj96.saas.bean.security.Token;

import javax.security.auth.login.LoginException;

/**
 * 认证业务接口
 */
public interface AuthService {

    /**
     * 用户名密码登录
     *
     * @param username 用户名
     * @param password 密码
     * @return token信息
     */
    Token login(String username,String password) throws LoginException;

    /**
     * 检查当前用户的token
     */
    Token check() throws LoginException;

    /**
     * 退出登录
     */
    void logout();

}

File: yangxj96-serve/yangxj96-serve-auth/src/main/kotlin/com/yangxj96/saas/server/auth/service/AuthService.kt
package com.yangxj96.saas.server.auth.service

import com.yangxj96.saas.bean.security.Token

/**
 * 认证业务接口
 */
interface AuthService {

    /**
     * 用户名密码登录
     *
     * @param username 用户名
     * @param password 密码
     * @return token信息
     */
    fun login(username: String, password: String): Token

    /**
     * 检查当前用户的token
     */
    fun check(): Token

    /**
     * 退出登录
     */
    fun logout()

}